### DISCUSSION TITLE
Dispositivi shelly senza internet

### DISCUSSION THREAD

**Discussion by disn3** [indomus/forum#2255] [GENERALE - Discussioni generiche]
Buonasera a tutti, vorrei il vostro aiuto per risolvere un problema legato al funzionamento di dispositivi shelly senza connessione internet. Mi spiego meglio: devo comandare dei dispositivi shelly (da remoto) attualmente collegati ad una connessione fibra fastweb. Voglio eliminare l'abbonamento fastweb dato che pago circa 30 euro al mese solo per comandare gli shelly ed ho quindi pensato ad un router wifi con sim, ma quello che possiedo permette la connessione di soli 10 dispositivi. C'è un modo (economico) per aumentare il numero di connessioni? Ho pensato anche ad home assistant che ho installato su raspberry pi 3, ma l'utilizzo da remoto senza connessione internet prevede il pagamento di un abbonamento. Mi consigliare un'alternativa ad HA gratuito che funziona raspberry pi 3 senza connessione internet?
Vi ringrazio anticipatamente

**Comment by NdK73** [comment]
Attenzione: il primo problema è che a parte pochissime eccezioni, le connessioni via rete cellulare non ti permettono l'accesso dall'esterno (almeno doppio NAT). Quindi questo problema è il primo che devi affrontare.
Le soluzioni più semplici:
1) usi il cloud Shelly e amen
2) Metti un RPi e acquisti il servizio NabuCasa
3) metti un RPi o simile e apri una VPN verso un tuo server (p.e. in un'altra casa dove arriva una connessione con IP pubblico)

La soluzione che ho adottato è una variante della 3 (con la VPN dei FritzBox ho "in omaggio" anche il routing tra le reti delle due case).

Per quanto riguarda il router, ho installato diversi anni fa da due TP-Link MR6400 che non hanno mai avuto particolari problemi. Recentemente, per avere un segnale migliore, ho montato un Mikrotik SXT (attaccandomi al palo della parabola sat :) ). La LAN la porto al Fritz 7590 (porta WAN, insieme alla FWA: in caso di problemi con la FWA mi basta cambiare gateway al 7590 e torno in linea).

**Comment by disn3** [comment]
Grazie per la risposta. Volendo optare per la 1 (uso cloud shelly e amen) utilizzando il router lte che possiedo (webcube 4), come mi consigli di raggirare il limite dei 10 dispositivi collegabili al router dato che gli shelly che dovrò agganciarci superano le 10 unità?

**Reply by blakdog1978** [reply]
Ma se metti un router con SIM perché dici senza internet? Economico cosa intendi ?

**Reply by disn3** [reply]
Sono due alternative, usare un RPi oppure una connessione con sim + cloud shelly. 
Con economico intendo adottare la seconda alternativa e sfruttare il router sim che possiedo (webcube4) apportando qualche modifica che permetta di aumentare il numero di 10 dispositivi da poter connettere al router stesso.

**Reply by blakdog1978** [reply]
Ciao se usi homeassistant puoi seguire la guida per connessioni dietro Nat e così sei raggiungibile dall'esterno, il tuo router non lo conosco, ma puoi mettere un extender wifi e gli fai fare a lui da server DHCP, io a casa ho un router 4G e poi mi sono fatto una rete mesh

**Reply by disn3** [reply]
Ho dato un'occhiata alla guida che mi hai suggerito, immagino che con questo sistema riesco a comunicare con homeassistant senza connessione internet ed evitando di acquistare il servizio NabuCasa... giusto?

**Reply by blakdog1978** [reply]
Giusto, per un po' è gratis poi finito il periodo di prova per il server su Google ti costerà sui 30 centesimi al mese

**Reply by snakuzzo** [reply]
il server su GCP non costa nulla, rispettando quanto previsto dal [free tier. di compute engine](https://cloud.google.com/free/docs/free-cloud-features#compute)
Solo se sfori il traffico mensile previsto (se non erro 1GB) puoi avere sovrapprezzi.

> 1 GB network egress from North America to all region destinations (excluding China and Australia) per month

**Reply by blakdog1978** [reply]
Ciao io sto ricevendo fatture che variano dai 18 ai 40 centesimi al mese... All'inizio era gratuito, magari ho sbagliato qualcosa io .... Comunque per quel che mi riguarda non mi dispiace pagarle...

**Reply by snakuzzo** [reply]
Io ho la VM attiva da mesi...sempre ZERO. Controlla la fattura e vedi dov'è che spendi

**Reply by disn3** [reply]
Usate entrambi raspberry?

**Reply by snakuzzo** [reply]
Anche...ma questo che c'entra?

**Reply by blakdog1978** [reply]
Compute Engine

**Reply by snakuzzo** [reply]
sì ok, ma nello specifico cosa ?
Compute Engine è la voce che racchiude tutto quello che riguarda la VM. I costi aggiuntivi potrebbero essere legati al traffico, CPU, dischi, backup, etc. ... controlla i dettagli
Guarda sotto `Billing > Cost Table`

![image](https://user-images.githubusercontent.com/7205371/211375484-d731ce38-8d2b-4e93-ad14-9bad94dccb20.png)

**Comment by blakdog1978** [comment]
Ecco

![fattura gcp](https://user-images.githubusercontent.com/36457031/211492641-618e039b-f4b3-4105-9297-fa5a7b3c331b.JPG)

**Reply by snakuzzo** [reply]
Eh ma devi vedere nei dettagli e fare un po' di somme...vedi che per alcune voci hai costo + sconto.

**Comment by disn3** [comment]
Quale router mi consigliate tra TP-Link Archer MR600 e FRITZ!Box 6820?

**Reply by KC-inDomus** [reply]
[FRITZ!](https://amzn.to/3jtBDgY) non foss'altro per la qualità complessiva di hardware e sistema operativo.
